lib/xclarity_client/connection/connection.rb

```diff
@@ -61,34 +59,37 @@ def do_post(uri = "", body = "")
     # @param [String] uri - endpoint to do the request
     # @param [JSON] body  - json to be sent in request body
     #
-    def do_put(uri = "", body = "")
-      @connection.put do |req|
-        req.url uri
-        req.headers['Content-Type'] = 'application/json'
-        req.body = body
-      end
-    rescue Faraday::Error::ConnectionFailed => e
-      $lxca_log.error "XClarityClient::XclarityBase do_put", "Error trying to send a PUT to #{uri}"
-      $lxca_log.error "XClarityClient::XclarityBase do_put", "Request sent: #{body}"
-      Faraday::Response.new
+    def do_put(uri = '', body = '')
+      build_request('put', uri, body)
     end
 
     #
     # Does a DELETE request to an LXCA endpoint
     #
     # @param [String] uri - endpoint to do the request
     #
-    def do_delete(uri = "")
-      @connection.delete do |req|
-        req.url uri
-        req.headers['Content-Type'] = 'application/json'
-      end
+    def do_delete(uri = '')
+      build_request('delete', uri)
     end
 
     private
 
+    def build_request(method, url, body = '')
```

Comment:
Receive the `method` as a symbol will be better here, once you don't will need to cast into a string